rust/tools/authorization-logic/build.rs

```diff
@@ -0,0 +1,54 @@
+// This is adapted from:
+// https://github.com/rrevenantt/antlr4rust/blob/master/build.rs
+
+use std::env;
+use std::error::Error;
+use std::process::Command;
+
+fn run_antlr_on(
+    grammar_file: &str,
+    antlr_path: &str
+    )-> Result<(), Box<dyn Error>> {
+
+    let parsing_subdir = env::current_dir().unwrap().join("src/parsing");
+    let full_file = grammar_file.to_owned() + ".g4";
+
+    // When debugging either this build script or the antlr, code it may be
+    // useful to build with
+    // 'cargo build -vv &> output file'
+    // because cargo omits errors generated from executed commands otherwise
+
+    Command::new("java")
+        .current_dir(parsing_subdir)
+        .arg("-jar")
+        .arg(antlr_path)
+        .arg("-Dlanguage=Rust")
+        .arg("-o")
+        .arg("antlr_gen")
+        .arg(&full_file)
+        .arg("-visitor") // This is not actually strictly needed
+        .spawn()
+        .expect("antlr4rust jar failed to start")
+        .wait_with_output()?;
+
+    println!("cargo:rerun-if-changed=src/parsing/{}.g4", grammar_file);
+
+    Ok(())
+}
+
+fn main() {
+
+    // At the moment, the jar for antlr is included locally. See these
+    // instructions to build it: https://github.com/rrevenantt/antlr4rust
+
+    let antlr_path = env::current_dir()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_owned()
```

Comment:
I like format better as well. Will do that.

The first unwrap is needed because env:current_dir() returns a `Result<PathBuf, Error>`. The second one is needed because [as this explains](https://stackoverflow.com/questions/37388107/how-to-convert-the-pathbuf-to-string) not all paths are UTF-8 encoded and so they can't all be converted into strings. As a result to_string on paths returns a `Result` and the second unwrap is needed. How about:

```
    let antlr_path = format!("{}/antlr4rust.jar",
                env::current_dir()
                .unwrap()
                .into_os_string()
                .into_string()
                .unwrap());
```